src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IUnaryOperatorExpression.vb

```diff
@@ -0,0 +1,1668 @@
+IUnaryOperatorExpression (UnaryOperationKind.IntegerBitwiseNegation) (OperationKind.UnaryOperatorExpression, Type: System.Int32)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of UnaryExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+
+        <Fact()>
+        Public Sub Test_UnaryOperatorExpression_Type_Not_System_UInt32()
+            Dim source = <![CDATA[
+Class A
+    Function Method() As System.UInt32
+        Dim i As System.UInt32 = Nothing
+        Return Not i'BIND:"Not i"
+    End Function
+End Class
+
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IUnaryOperatorExpression (UnaryOperationKind.IntegerBitwiseNegation) (OperationKind.UnaryOperatorExpression, Type: System.UInt32)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.UInt32)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of UnaryExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+
+        <Fact()>
+        Public Sub Test_UnaryOperatorExpression_Type_Not_System_Int64()
+            Dim source = <![CDATA[
+Class A
+    Function Method() As System.Int64
+        Dim i As System.Int64 = Nothing
+        Return Not i'BIND:"Not i"
+    End Function
+End Class
+
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IUnaryOperatorExpression (UnaryOperationKind.IntegerBitwiseNegation) (OperationKind.UnaryOperatorExpression, Type: System.Int64)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int64)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of UnaryExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+
+        <Fact()>
+        Public Sub Test_UnaryOperatorExpression_Type_Not_System_UInt64()
+            Dim source = <![CDATA[
+Class A
+    Function Method() As System.UInt64
+        Dim i As System.UInt64 = Nothing
+        Return Not i'BIND:"Not i"
+    End Function
+End Class
+
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IUnaryOperatorExpression (UnaryOperationKind.IntegerBitwiseNegation) (OperationKind.UnaryOperatorExpression, Type: System.UInt64)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.UInt64)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of UnaryExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+
+        <Fact()>
+        Public Sub Test_UnaryOperatorExpression_Type_Not_System_Boolean()
+            Dim source = <![CDATA[
+Class A
+    Function Method() As System.Boolean
+        Dim i As System.Boolean = Nothing
+        Return Not i'BIND:"Not i"
+    End Function
+End Class
+
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IUnaryOperatorExpression (UnaryOperationKind.BooleanBitwiseNegation) (OperationKind.UnaryOperatorExpression, Type: System.Boolean)
```

Comment:
There should be some. From language spec, section **11.17 Logical Operators**:

> The And, Not, Or, and Xor operators, which are called the logical operators, are evaluated as follows: ...

However, we should figure out how do they map to IOperation universe.